### ISSUE TITLE
<dateRange>: best practice

### ISSUE THREAD

**Issue by SJagodzinski** [SAA-SDT/eac-cpf-schema#32]
Describe best practice for unknown end dates in `<toDate>` within `<dateRange>`.

## Creator of issue
1. Silke Jagodzinski
2. TS-EAS: EAC-CPF subgroup
3. s.jagodzinski@bundesarchiv.de or silkejagodzinski@gmail.com

## The issue relates to
- [ ] EAC-CPF schema issue
- [x] EAC-CPF Tag Library issue
- [ ] EAD schema issue
- [ ] EAD Tag Library issue
- [ ] Schema issue
- [ ] Tag Library issue
- [ ] Suggestions for all schemas
- [ ] Suggestions for all Tag Libraries
- [ ] Other

## Wanted change/feature with context
> `<fromDate>` and `<toDate>` within `<dateRange>` are mandatory according to the TL but not in the schema. Please correct the schema.

[APEx project, May 2013]

Comment Anila Anjeli:
`<toDate>` is not mandatory in the schema. One should be able to record an “open” range, i.e. containing only fromDate, yes? 

> Using `<dateRange>` with a `<toDate>` not known. `<toDate>` is mandatory in TL (and schema). When not having a `<toDate>` one solution used is to put 2099 as the year. Some advice on how to use `<dateRange>` when no `<toDate>` is known would be appreciated. Often when describing a `<corporateBody>` people want to use a `<dateRange>` instead of a `<date>` that should be the way to do it. A clarification on the use in the TL could be good.

[APEx, May 2013]

Comment Anila Anjeli:
While `<toDate>` is not mandatory in the schema, the second issue raised here needs discussion.

Comment unknown: 
ONE OR THE OTHER OR BOTH. Cannot be enforced formally. Tag library needs to express this as a best practice (no empty date ranges).

Comment TS-EAC so far
All of the proposals accepted

**Comment by karinbredenberg** [comment]
Can the work of the extended datetime group at LoC be of a help? There are work going on with ISO datetime which might be the solution. the work from extended datetime is incooperated in the iso work.

**Comment by karinbredenberg** [comment]
Having problems with this one right now. How todo when you dont have the toDate? First problem is the discrepancy as noted between the TL and the schema. In EAD3 todate is optional. So perhaps reconciliation with EAD and follow how the EAC-CPF schema is made?

**Comment by karinbredenberg** [comment]
Just a an update. To make it smooth in our use we are following the schema and not the TL here.

**Comment by SJagodzinski** [comment]
current eac schema: `<fromDate>` & `<toDate>` occur only within `<dateRange>` and both are optional.
current eac TL: `<fromDate>` & `<toDate>` occur only within `<dateRange>` and both are mandatory.
ead3 schema: `<fromdate>` & `<todate>` occur only within `<daterange>` and both are optional.

Following the schema implies to change TL description and add best practise for unknown end dates.
Changing TL description seems clear and easy, but what is the best practise here?

**Comment by karinbredenberg** [comment]
Usually you have todate optional or you have an text saying "open" or similar. But having a word means it needs to be changed from a date to a mixed format with our own validations so just not having it there is in my opinion a better solution.
Most often whats wanted is to say its a range but I don't know when it stops instead of just having one date and adding the local attribute saying its a start date. So in the dateRange add text saying if the to Date is unknown omit the toDate element.

**Comment by wisserk** [comment]
Change the tag library so that it is consistent with the schema and with EAD3 implementation.

**Comment by wisserk** [comment]
Results of team discussion: 

Change the tag library so that it is consistent with the schema and with EAD3 implementation.

Discussion points:
Ranges with unknown dates make conceptual sense (e.g., a life range is still a range even if one of the dates is unknown). 

Encoding and presentation both need to be taken into consideration; encoding is the business of the standard and presentation is locally controlled. 

**Best practices and the potential role of documentation needs to be readdressed in Phase 2.**

**Comment by SJagodzinski** [comment]
The updated TL contains the sentence 

> If the <toDate> is not known, it may be omitted.

for both elements //fromDate and //toDate.

This issues will be discussed in phase 2, if there is a need for more best practise explanations.

**Comment by ailie-s** [comment]
Outcomes of discussion in Austin, August 2019:

1. Update ‘Description and Usage’ section of EAC-CPF Tag Library to include information about date ranges with either no start date or no end date. It should state that it is expected that a `<dateRange>` will have at least one of `<fromDate>` or `<toDate>`, but that one or other may be omitted if the information is not known.

2. Add optional `@status` attribute to indicate where dates are unknown or ongoing.
Examples:
```
<dateRange>
    <fromDate status="unknown"/>
    <toDate standardDate="1832-03-29">29 March 1832</toDate>
</dateRange>
```

```
<dateRange>
    <fromDate standardDate="1762-09-19">19 September 1762</fromDate>
    <toDate status="unknown"/>
</dateRange>
```

```
<dateRange>
    <fromDate standardDate="1853">1853</fromDate>
    <toDate status="open"/>
</dateRange>
```

**Comment by SJagodzinski** [comment]
Discussed in #70 and solved with #254 : new attribute `@status` used to indicate unknown and ongoing dates.
